### PR TITLE
Use path and componentId as map keys

### PR DESCRIPTION
**Pull request by tamorim**
At least for the case of `ExhibitionMode` component, makes navigation work much better!

### CODE REVIEW

**Comment by brenoc** [comment]
`result` should be kept, because it's useful to handle AJAX errors.

**Comment by tamorim** [comment]
@brenoc Updated!

File: src/stores/SearchStore.js [hunk 1]
```diff
@@ -3,13 +3,15 @@ import immutable from 'alt/utils/ImmutableUtil';
...
   return state.withMutations(map => {
     for (let componentId in products) {
-      let results = Immutable.Map({ results: pluck(products[componentId], 'slug') });
-      map.set(componentId, results);
+      let productSlug = pluck(products[componentId], 'slug');
+      let results = Immutable.Map().setIn([componentId, 'results'], productSlug);
+      map.set(path, results);
     }
     for (let componentId in product) {
       let results = Immutable.Map({ results: pluck(product[componentId], 'slug') });
```
Comment (brenoc) [inline src/stores/SearchStore.js]: Should update this as well

**Comment by tamorim** [comment]
@brenoc Updated!

**Pull request merged**